parsec/class/parsec_rbtree.c

```diff
@@ -0,0 +1,331 @@
+#include <stdbool.h>
+
+#include "parsec/parsec_config.h"
+
+#include "parsec/class/parsec_rbtree.h"
+#include "parsec/constants.h"
+
+/* left and right children, using list pointers */
+#define LEFT(node)  (*(parsec_rbtree_node_t**)&node->super.list_prev)
+#define RIGHT(node) (*(parsec_rbtree_node_t**)&node->super.list_next)
+
+
+/**
+ * The list_item object instance.
+ */
+static inline void
+parsec_rbtree_node_construct( parsec_rbtree_node_t* item )
+{
+    item->color = PARSEC_RBTREE_BLACK;
+}
+
+PARSEC_OBJ_CLASS_INSTANCE(parsec_rbtree_node_t, parsec_list_item_t,
+                          parsec_rbtree_node_construct, NULL);
+
+void parsec_rbtree_init(parsec_rbtree_t* tree, size_t offset) {
+    tree->nil = PARSEC_OBJ_NEW(parsec_rbtree_node_t);
```

Comment:
Instead of allocating the `nil` you can make it point to the `tree` itself. The code only uses it to compare for end, of branches, so it should work just fine.